### PR TITLE
Ncl 6290 expose metrics from kafka store

### PR DESCRIPTION
**Pull request by lbossis**
@avibelli

### CODE REVIEW

File: src/main/java/org/jboss/pnc/kafkastore/common/SortedSetOrder.java [hunk 1]
```diff
@@ -42,6 +43,7 @@
 
     ArrayList<String> sorted = new ArrayList<>();
 
+    @Timed
```
Comment (vibe13) [inline src/main/java/org/jboss/pnc/kafkastore/common/SortedSetOrder.java:46]: Mmmm this is not an exposed REST enpdoint, so not sure whether @Timed is applicable. Also, not part of micrometer spec.

File: src/main/java/org/jboss/pnc/kafkastore/rest/MainRest.java [hunk 2]
```diff
@@ -33,6 +35,7 @@
 
 @Path("/")
 @ApplicationScoped
+@Timed
```
Comment (vibe13) [inline src/main/java/org/jboss/pnc/kafkastore/rest/MainRest.java:38]: From https://quarkus.io/guides/micrometer#writing-the-application, it looks like ```Timer``` and `Counter` are automatically exposed for all the endpoints, no need to specify them.
Also, please verify in https://quarkus.io/guides/micrometer#configuration-reference if the defaults are enough for us, or we need to add some configurations.

File: src/main/java/org/jboss/pnc/kafkastore/mapper/BuildStageRecordMapper.java [hunk 3]
```diff
@@ -90,4 +106,18 @@ private String buildId(String processContext) {
             return processContext;
         }
     }
+
+    @GET
+    @Produces(MediaType.TEXT_PLAIN)
```
Comment (vibe13) [inline src/main/java/org/jboss/pnc/kafkastore/mapper/BuildStageRecordMapper.java:111]: I think it's good to track errors. But this is not an endpoint, it's a mapper. You need to move these endpoints to the MainRest class.

File: pom.xml [hunk 4]
```diff
@@ -103,6 +103,18 @@
             <groupId>io.quarkus</groupId>
             <artifactId>quarkus-smallrye-reactive-messaging-kafka</artifactId>
         </dependency>
+        <dependency>
```
Comment (vibe13) [inline pom.xml:106]: I am reading from https://quarkus.io/guides/micrometer that Micrometer is the recommended approach to metrics for Quarkus. Could you please switch the implementation to use micrometer instead? You should probably only add these 2 dependencies:
```
<dependency>
    <groupId>io.quarkus</groupId>
    <artifactId>quarkus-micrometer</artifactId>
</dependency>
<dependency>
    <groupId>io.micrometer</groupId>
    <artifactId>micrometer-registry-prometheus</artifactId>
</dependency>
```

File: src/main/java/org/jboss/pnc/kafkastore/facade/BuildMetricsFetcher.java [hunk 5]
```diff
@@ -28,6 +29,7 @@
 import java.util.List;
 
 @ApplicationScoped
+@Timed
```
Comment (vibe13) [inline src/main/java/org/jboss/pnc/kafkastore/facade/BuildMetricsFetcher.java:32]: Same as above

File: src/main/java/org/jboss/pnc/kafkastore/mapper/BuildStageRecordMapper.java [hunk 6]
```diff
@@ -20,18 +20,32 @@
...
+import javax.ws.rs.core.MediaType;
 import java.io.IOException;
 import java.util.Optional;
 
 @ApplicationScoped
+@Timed
 @Slf4j
 public class BuildStageRecordMapper {
 
+    static final Counter exceptionsTotal = Counter.build()
```
Comment (vibe13) [inline src/main/java/org/jboss/pnc/kafkastore/mapper/BuildStageRecordMapper.java:43]: You can probably use a Gauge (2 Gauges) for these counters, according to examples in https://quarkus.io/guides/micrometer

**Review by vibe13 — CHANGES_REQUESTED** [review]
@lbossis Hi Leo, I might have put you on the wrong path. From the docs, it looks like the micrometer way is recommended for Quarkus-based applications (and it looks easier!). Would you please switch to Micrometer spec please? Thanks!

**Comment by lbossis** [comment]
Implementation was switched from microprofile to micrometer. This PR will be closed without merge.

**Pull request closed without merging**